ocaml/xapi/xapi_pif.ml

```diff
@@ -796,25 +797,41 @@ let rec unplug ~__context ~self =
   if Db.PIF.get_capabilities ~__context ~self |> List.mem "fcoe" then
     assert_fcoe_not_in_use ~__context self;
 
-  let tunnel = Db.PIF.get_tunnel_transport_PIF_of ~__context ~self in
+  let tunnel = pif_rec.API.pIF_tunnel_transport_PIF_of in
```

Comment:
```ocaml
match pif_rec.API.pIF_tunnel_transport_PIF_of with
| tunnel :: _ ->
     debug "PIF is tunnel transport PIF... also bringing down access PIF";
     let access_PIF = Db.Tunnel.get_access_PIF ~__context ~self:tunnel in
     unplug ~__context ~self:access_PIF;
| _ -> ();
```
or even
```ocaml
List.iter (fun tunnel -> 
     debug "PIF is tunnel transport PIF... also bringing down access PIF";
     let access_PIF = Db.Tunnel.get_access_PIF ~__context ~self:tunnel in
     unplug ~__context ~self:access_PIF) 
     pif_rec.API.pIF_tunnel_transport_PIF_of;
```

Reply:
Got it. I like the latter. Indeed this looks much better.